libs/core/src/lib/action-sheet/action-sheet-body/action-sheet-body.component.ts

```diff
@@ -71,6 +72,15 @@ export class ActionSheetBodyComponent {
     @HostListener('keydown', ['$event'])
     keyDownHandler(event: KeyboardEvent): void {
         if (this._keyboardSupportService.keyManager) {
+            if (KeyUtil.isKeyCode(event, TAB)) {
+                event.preventDefault();
+                if (hasModifierKey(event, 'shiftKey')) {
+                    this._keyboardSupportService.keyManager.setPreviousItemActive();
+                } else {
+                    this._keyboardSupportService.keyManager.setNextItemActive();
+                }
+                return;
+            }
```

Comment:
Move this to the keyboardService.onKeyDown like this:

```ts
onKeyDown(event: KeyboardEvent): void {
        this._keyManager.onKeydown(event);
        if (KeyUtil.isKeyCode(event, TAB)) {
            event.preventDefault();
            hasModifierKey(event, 'shiftKey')
                ? this._keyManager.setPreviousItemActive()
                : this._keyManager.setNextItemActive();
        }
    }
```

This way not only here, but everywhere this issue will be resolved, Problem generally is that on TAB ListKeyManager does not update activeItem, it just relies on native TAB behavior